packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -301,6 +327,27 @@ class LinearProgressIndicator extends ProgressIndicator {
   /// {@endtemplate}
   final double? minHeight;
 
+  /// The border radius of the widget.
+  ///
+  /// By default it is [BorderRadius.zero], which produces a rectangular shape.
+  final BorderRadiusGeometry outerBorderRadius;
+
+  /// The border radius of the indicator.
+  ///
+  /// When TextDirection is set to [TextDirection.ltr], [BorderRadius.topLeft]
```

Comment:
Why override the radius? There will always be a text direction.

Reply:
This is what I expect, so I tried to override the start corner to always be 0. What do you think? Else, BorderRadius.circular(32) will always output a weird unwanted value, which will force everybody to do BorderRadius(topLeft, topRight) which is a lot more verbose (even more if the person is considering accessibility / Directionality, which I'm already handling for them for free).

![expected](https://user-images.githubusercontent.com/351125/232666620-eb3f506a-3a7b-4965-9e22-b490cf9bdc3e.png)

Reply:
Much like the circular PR, I think we should not try to assume how folks will and won't want to use this. Why expose it if we are going to override it?

Reply:
We want to expose the end-side, not the start-side. We could just expose `Radius` instead of `BordeRadius`, but could be slightly odd since all the rest of the framework uses BorderRadius. What do you think?

Reply:
> We want to expose the end-side, not the start-side

Why? Folks have the option to apply the radius to the track with this PR.

Reply:
Could you provide a single example in any software where the linear indicator starts rounder than the end? I've never seen one. I think we should make it easy for people, not ask for 5 LOC when it could be 1. But I might be wrong. What if I took this to Reddit? Could I ask there for the feedback of the people what they prefer?

Reply:
> Could you provide a single example in any software where the linear indicator starts rounder than the end? I've never seen one.

No. That is not the point.

> What if I took this to Reddit? 

You're welcome too. I don't feel that this kind of response is appropriate in the context of code review though. I am making a point about a clear API. I feel like it is unfortunate that we expose this and then ignore it.

I also do not think it is worth arguing over based purely on the fact that you personally do not see value in it like we did with the circular PR, and going back and forth over it like in that PR is an unfortunate waste of both our time.

Reply:
I think we are discussing about introducing a feature that wouldn't be useful. That's part of the Yak Shaving in Flutter documentation "avoid implementing what you don't need". I don't think we need a start-radius indicator, and I think we could "crowd-source" to see how others feel.

But... just like the other, yes, I can ignore this and we can have the "weird" behavior by default on Flutter. It is just unfortunate for me, that everybody will need boilerplate to get what they want.

Reply:
New proposal: single borderRadius for everything.